buildSrc/src/main/groovy/org/elasticsearch/gradle/test/RestIntegTestTask.groovy

```diff
@@ -248,7 +249,11 @@ class RestIntegTestTask extends DefaultTask {
             restSpec
         }
         project.dependencies {
-            restSpec project.project(':rest-api-spec')
+            if (ClasspathUtils.isElasticsearchProject()) {
```

Comment:
I don't think this will work when running tests via the `:build-tools` project as this will be considered "external". This will only be true in build logic, or when running the task in `:buildSrc`. We'll need to either explicitly add the `buildSrc` maker file to the test classpath here, or include some other flag to tell this utility it's running within the Elasticsearch build.